cpp/mrc/tests/test_edges.cpp

```diff
@@ -842,4 +842,41 @@ TEST_F(TestEdges, CreateAndDestroy)
     }
 }
 
+TEST_F(TestEdges, EdgeTapWAcceptorWProvider)
+{
+    auto source = std::make_shared<node::TestSource<int>>();
+    auto node   = std::make_shared<node::TestNode<int>>();
+    auto sink   = std::make_shared<node::TestSink<int>>();
+
+    // Original edge
+    mrc::make_edge(*source, *sink);
+
+    // Tap edge
+    mrc::edge::EdgeBuilder::make_edge_splice<int>(*source, *sink, *node, *node);
```

Comment:
This seems odd to me. Why do we need to specify `*node` twice here? And shouldnt we be able to deduce the edge type if they are all the same `<int>`?

Reply:
Node is what is used to deduce if its an Acceptor or Provider, the EdgeBuilder API for splice edge doesn't assume that the SpliceInput and the SpliceOutput are the same node. One reason for this is that, one could be the input to a module, and one could be the output of the module, in which case they won't be the same node.